Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -0,0 +1,348 @@
+  satisfied through some component other than Linux (more on this later).
+
+  There are several additional reasons for favouring an external component, besides
+  what has been mentioned already:
+  *  much greater probability for the component to be simpler (and thus easier to qualify)
+  *  opportunity to have a better control over the component (e.g. use a software
+     architecture that is better suited for safety, formalised processes, etc.)
+  *  decouple release cycles of the Linux kernel from the safety component
+  *  other non-technical reasons (e.g. different licensing)
+
+## **Examples of Safety-oriented System Design Concepts**
+
+**Premise:** applying the initial consideration from the previous sections, it is
+necessary to define what is required at system level in terms of both availability
+and integrity.
+
+The simpler - and most common - case is to require detection of interference
+affecting components that are recipient of safety requirements.
+
+Interference affecting a component with safety requirements must be both detected
+and reported, in a way that is compatible with the associated time requirements.
+At this point, appropriate action(s) will be taken, in accordance with the
+pre-established policy, to ensure adequate availability
+(e.g. restarting, onlining of backup system, etc.)
+
+As mentioned earlier, guaranteeing availability can be much more difficult
+and/or costly.
+
+* **Application Monitoring**
+
+  Monitoring is a simple approach to detection of interference that affects
+  component with safety requirement.
+  
+  It can save from extensive hardening.
+
+  However two conditions must be met:
+  1. It must be possible to express a simple check, in a way that can repeatedly,
+     reliably and unequivocally produce a YES/NO verdict about the health parameter
+     of interest
+  3. The observation whether the condition is met (or not) can be performed from a
+     context having same or better safety qualification.
+     And the context is able to take whatever action might be dictated by the
+     specific policy in place.
+
+  Assuming that the these two conditions are met (e.g. the system contains either a
+  watchdog or an equivalent component -- more on this later) then the opportunistic
+  approach is to identify an high-level activity that can be tied to the monitor.
+
+  A very crude example of this validation could be to require that application
+  responsible of performing safety relevant operations would ping a watchdog,
+  periodically.
+
+  As long as the application pings the watchdog within the set deadline, the watchdog
+  does not trigger any exceptional operation.
+
+  The example is crude, but it illustrates how - for example - all the internal linux
+  components involved with booting, loading and executing the application are not
+  subject to any safety requirement, because only the final product of their operation
+  is monitored.
```

Comment:
In the period above you are saying that booting, loading, executing the app are not subject to any safety requirements but you have not described why. So it is too early for a conclusion like this here. Instead I would say:

The example is simple, but it suggests how - for example - all the HW/SW stack involved with booting, loading and executing the application could be
  monitored against time interference deriving from any Linux failure mode or failure modes of the HW supporting the Linux execution.

Reply:
"...components involved with booting, loading and executing the application might arguably not need to be subject to any safety requirement, provided that the final product of their operation can be monitored to ensure that it satisfies the required criteria."